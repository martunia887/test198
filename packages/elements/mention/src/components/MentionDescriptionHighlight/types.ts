import { MentionDescription } from '../../types';

export interface DescriptionHighlightProps {
  mention: MentionDescription;
}
