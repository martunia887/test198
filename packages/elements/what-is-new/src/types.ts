export interface Feature {
  date: number;
  title: string;
  description: string;
  link: string;
}
