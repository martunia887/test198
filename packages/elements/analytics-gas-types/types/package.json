{"name":"@atlaskit/analytics-gas-types/types","main":"../dist/cjs/types.js","module":"../dist/esm/types.js","types":"../dist/esm/types.d.ts"}