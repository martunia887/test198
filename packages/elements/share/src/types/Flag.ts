export const OBJECT_SHARED = 'object-shared';

export const ADMIN_NOTIFIED = 'admin-notified';

export type FlagType = 'object-shared' | 'admin-notified';

export type Flag = {
  id: string;
  type: FlagType;
  localizedTitle: string;
};
