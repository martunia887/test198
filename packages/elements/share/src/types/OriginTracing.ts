export type OriginTracingWithIdGenerated = {
  originIdGenerated: string;
  originProduct: string;
};

export type OriginTracingForSubSequentEvents = {
  originId: string;
  originProduct: string;
};

export type OriginTracing = {
  id: string;
  addToUrl: (link: string) => string;
  toAnalyticsAttributes: (
    args: { hasGeneratedId: boolean },
  ) => OriginTracingWithIdGenerated | OriginTracingForSubSequentEvents;
};

export type OriginTracingFactory = () => OriginTracing;
