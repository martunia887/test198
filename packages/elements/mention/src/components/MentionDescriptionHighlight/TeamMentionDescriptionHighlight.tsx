import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import { DescriptionHighlightProps } from './types';
import { DescriptionHighlightStyle } from './styles';
import { messages } from '../i18n';

export default class TeamMentionDescriptionHighlight extends React.PureComponent<
  DescriptionHighlightProps,
  {}
> {
  renderTeamInformation(memberCount?: number, includesYou?: boolean) {
    // TEAMS-333 This same logic is implemented in User Picker as well. https://product-fabric.atlassian.net/browse/TEAMS-333
    // will extract this to a common place

    // if Member count is missing, do not show the byline, regardless of the availability of includesYou
    if (typeof memberCount !== 'number') {
      return null;
    }

    return (
      <DescriptionHighlightStyle>
        <FormattedMessage
          {...(memberCount > 50
            ? messages.plus50Members
            : messages.memberCount)}
          values={{ count: memberCount, includes: includesYou }}
        />
      </DescriptionHighlightStyle>
    );
  }

  render() {
    const { context } = this.props.mention;

    const includesYou = context && context.includesYou;
    const memberCount = context && context.memberCount;

    return this.renderTeamInformation(memberCount, includesYou);
  }
}
