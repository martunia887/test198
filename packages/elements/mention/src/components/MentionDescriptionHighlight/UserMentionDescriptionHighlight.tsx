import * as React from 'react';
import { DescriptionHighlightProps } from './types';
import { renderHighlight } from '../MentionItem/MentionHighlightHelpers';
import { DescriptionHighlightStyle } from './styles';

export default class UserMentionDescriptionHighlight extends React.PureComponent<
  DescriptionHighlightProps,
  {}
> {
  render() {
    const { highlight, nickname } = this.props.mention;
    const nicknameHighlights = highlight && highlight.nickname;

    return renderHighlight(
      DescriptionHighlightStyle,
      nickname,
      nicknameHighlights,
      '@',
    );
  }
}
