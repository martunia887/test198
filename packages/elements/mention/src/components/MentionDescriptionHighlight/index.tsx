import * as React from 'react';
import { UserType } from '../../types';
import { DescriptionHighlightProps } from './types';
import UserMentionDescriptionHighlight from './UserMentionDescriptionHighlight';
import TeamMentionDescriptionHighlight from './TeamMentionDescriptionHighlight';

export default class MentionDescriptionHighlight extends React.PureComponent<
  DescriptionHighlightProps,
  {}
> {
  render() {
    const { userType } = this.props.mention;

    switch (userType) {
      case UserType[UserType.TEAM]: {
        return <TeamMentionDescriptionHighlight mention={this.props.mention} />;
      }
      default: {
        return <UserMentionDescriptionHighlight mention={this.props.mention} />;
      }
    }
  }
}
