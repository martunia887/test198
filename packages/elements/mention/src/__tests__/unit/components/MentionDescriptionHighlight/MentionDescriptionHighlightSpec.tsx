import { shallow } from 'enzyme';
import * as React from 'react';
import { MentionDescription } from '../../../../types';
import MentionDescriptionHighlight from '../../../../components/MentionDescriptionHighlight';
import { userMention, teamMention } from './_commonData';

const shallowRender = (mention: MentionDescription) =>
  shallow(<MentionDescriptionHighlight mention={mention} />);

describe('Mention Description Highlight', () => {
  it('should render User Mention description if a user is provided', () => {
    const component = shallowRender(userMention);
    expect(component).toMatchSnapshot();
  });

  it('should render Team Mention description if a team is provided', () => {
    const component = shallowRender(teamMention);
    expect(component).toMatchSnapshot();
  });
});
