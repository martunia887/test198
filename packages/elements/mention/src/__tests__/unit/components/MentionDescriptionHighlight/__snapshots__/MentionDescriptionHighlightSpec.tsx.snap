// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mention Description Highlight should render Team Mention description if a team is provided 1`] = `
<TeamMentionDescriptionHighlight
  mention={
    Object {
      "accessLevel": "CONTAINER",
      "avatarUrl": "www.example.com/image.png",
      "context": Object {
        "includesYou": true,
        "memberCount": 5,
        "members": Array [
          Object {
            "id": "user-1234",
            "name": "Test User",
          },
        ],
      },
      "id": "12345",
      "name": "Test team",
      "userType": "TEAM",
    }
  }
/>
`;

exports[`Mention Description Highlight should render User Mention description if a user is provided 1`] = `
<UserMentionDescriptionHighlight
  mention={
    Object {
      "accessLevel": "CONTAINER",
      "avatarUrl": "www.example.com/image.png",
      "id": "12345",
      "nickname": "Test User",
      "userType": "APP",
    }
  }
/>
`;
